app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt

```diff
@@ -157,6 +164,26 @@ class WorkoutProviderImpl : WorkoutProvider {
         workoutRepo.updateTemplateWorkout(workoutId, date, newExercises)
     }
 
+    /**
+     * Queries the past workouts and converts it to a [HistoryInfoWorkout]
+     * @param workoutId
+     * @see HistoryInfoWorkout
+     * @see toHistoryInfoWorkout
+     */
+    override suspend fun setClickedHistoryWorkout(workoutId: String) {
+        getPastWorkouts().take(1).collect { workouts ->
+            workouts.find { workout -> workout.id == workoutId }?.let { result ->
+                _clickedPastWorkout.value = result.toHistoryInfoWorkout()
+            }
+        }
+    }
+
+    /**
+     * Returns the last workout that has been clicked on
+     * @return [HistoryInfoWorkout]
+     */
+    override fun getClickedHistoryWorkout(): Flow<HistoryInfoWorkout> = clickedPastWorkout
```

Comment:
You can put the flow field in the interface instead of using a function to expose it.

Reply:
The same is valid for the other private flows. You can declare them in the interface and use a `private set()` for each of the overrides and set them inside this class accordingly.

Reply:
The other fields are ready, however I spoke with @Gotalicp and we decided to make it in another pr after the [https://github.com/HauntedMilkshake/fitness_app/pull/48](url) is merged